cli/test-utils/src/macros.rs

```diff
@@ -77,6 +77,122 @@ macro_rules! forgetest_init {
     };
 }
 
+/// Clones an external repository and makes sure the tests pass.
+#[macro_export]
+macro_rules! forgetest_external {
+    ($test:ident, $repo:literal) => {
+        $crate::forgetest_external!($test, $repo, $crate::ethers_solc::PathStyle::Dapptools);
+    };
+    ($test:ident, $repo:literal, $style:expr) => {
+        #[test]
+        fn $test() {
+            use std::process::{Command, Stdio};
+            let (prj, mut cmd) = $crate::util::setup(stringify!($test), $style);
+
+            // Wipe the default structure
+            prj.wipe();
+
+            // Clone the external repository
+            let git_clone = Command::new("git")
+                .args([
+                    "clone",
+                    "--depth",
+                    "1",
+                    "--recursive",
+                    &format!("https://github.com/{}", $repo),
+                    prj.root().to_str().expect("could not get project root"),
+                ])
+                .stdout(Stdio::null())
+                .stderr(Stdio::null())
+                .status()
+                .expect("could not clone repository. is git installed?");
+            assert!(git_clone.success(), "could not clone repository");
+
+            // We just run make install, but we do not care if it worked or not
+            let make_install = Command::new("make")
+                .arg("install")
+                .current_dir(prj.root())
+                .stdout(Stdio::null())
+                .stderr(Stdio::null())
+                .status();
+
+            // Run the tests
+            cmd.arg("test")
+                .arg("--optimize")
+                .arg("--optimize-runs")
+                .arg("20000")
+                .arg("--ffi")
+                .set_env("FOUNDRY_FUZZ_RUNS", "1");
+            cmd.assert_non_empty_stdout();
+        }
+    };
+}
+
+/// Like forgetest_external! but with RPC forking
+#[macro_export]
+macro_rules! forgetest_external_forking {
+    ($test:ident, $repo:literal, $fork_block:literal) => {
```

Comment:
100%, I just opened the draft because the tests were failing to run on my computer (see 19c0d048868ff6ef363b412ad9fa691ba26a99b7) and I asked Brock for help :sweat_smile: